### ISSUE TITLE
Bookmarklet for testing character key shortcuts  (SC 2.1.4)

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#833]
I'd be happy to get feedback about a bookmarklet that we are using to test 2.4.1. 
I have created a test page to demonstrate how it works:
http://3needs.org/en/testing/code/kb-shortcuts.html

**Comment by patrickhlauke** [comment]
sorry, VERY late comment on this (as i'm digging through the mouldy parts of my github notification emails folder) ...

might be worth clarifying in the description page there when you say

> When the bookmarklet script reaches 'p' and 'P', an alert pauses the execution of the script. In real word cases, the script should now be checked to see whether any unexpected / disruptive functions are triggered.

that the alerts that pause the execution are dummy code in the test page itself - i.e. it's not the bookmarklet that noticed something was caught/intercepted by the page, but it's the page reacting to the synthetic key press.

in practice, this bookmarklet would trigger any functionality in fast succession - so if it wasn't an execution-breaking script in the test page, it would trigger the stuff and then carrying on triggering more stuff? i.e. if i fire off the bookmarklet on a youtube video page, it'll trigger all sorts of things one after the other really quickly

**Comment by patrickhlauke** [comment]
randomly...can this issue be closed? (I just stumbled across it mainly because I was looking for the URL of the bookmarklet :) )

**Comment by detlevhfischer** [comment]
Well, if there was more discussion to come, I'd like to keep this open. Just today I came across a page where a tester had identified two single key keyboard shortcuts that activate behaviour on the page under test https://www.demokratie-leben.de/das-programm/ueber-demokratie-leben
the m key (mutes video) and the f key (switches to fullscreen view). At least the latter intercepts screen reader commands (NVDA: next form field). The bookmarklet seems unable to detect this, it doesn't flag m and f keys as violating 2.1.4...

**Comment by patrickhlauke** [comment]
this has been open since July 2019, so not sure what more feedback you're expecting ;)

might be worth putting the bookmarklet somewhere like github and allowing people to submit issues/PRs there?

**Comment by JAWS-test** [comment]
@detlevhfischer 

In your example f and m only work as shortcuts within the video. I assume that a video is an interface component and therefore the exception is valid:

> Active only on focus:    The keyboard shortcut for a user interface component is only active when that component has focus.

**Comment by detlevhfischer** [comment]
@JAWS-test Good point. BTW the bookmarklet does catch the first key (f) when the video running before the bookmarklet is triggered.
